lib/internal/vm.js

```diff
@@ -97,22 +101,65 @@ function internalCompileFunction(code, params, options) {
   }
 
   if (importModuleDynamically !== undefined) {
-    validateFunction(importModuleDynamically,
-                     'options.importModuleDynamically');
-    const { importModuleDynamicallyWrap } = require('internal/vm/module');
-    const wrapped = importModuleDynamicallyWrap(importModuleDynamically);
-    const func = result.function;
-    const { registerModule } = require('internal/modules/esm/utils');
-    registerModule(func, {
-      __proto__: null,
-      importModuleDynamically: wrapped,
-    });
+    registerImportModuleDynamically(result.function, importModuleDynamically);
   }
 
   return result;
 }
 
+function makeContextifyScript(code,
+                              filename,
+                              lineOffset,
+                              columnOffset,
+                              cachedData,
+                              produceCachedData,
+                              parsingContext,
+                              hostDefinedOptionId,
+                              importModuleDynamically) {
+  let script;
+  // Calling `ReThrow()` on a native TryCatch does not generate a new
+  // abort-on-uncaught-exception check. A dummy try/catch in JS land
+  // protects against that.
```

Comment:
should the try/catch be done inside `ContextifyScript`, then, so that callsites don't need to repeat it?

Reply:
@ljharb I think this is from https://github.com/nodejs/node/pull/50137, not from this PR.